### PR TITLE
CB-21089 Configure salt master worker threads

### PR DESCRIPTION
**Pull request by foldik**
Configure salt master worker threads during. The motivation is to add more threads to salt master for better performance on bigger instances. This takes effect during cluster creation or during node repair when the user repairs the node where salt master is running.

### CODE REVIEW

File: freeipa/src/main/java/com/sequenceiq/freeipa/service/BootstrapService.java [hunk 1]
```diff
@@ -129,4 +133,72 @@ private String getHostname(FreeIpa freeIpa, InstanceMetaData im) {
...
+            Credential credential = credentialService.getCredentialByEnvCrn(stack.getEnvironmentCrn());
+            ExtendedCloudCredential extendedCloudCredential = convertCredentialToExtendedCloudCredential(credential, cloudPlatform);
+            Optional<Integer> cpuCountForInstanceType =
+                    pricingCache.get().getCpuCountForInstanceType(stack.getRegion(), masterInstanceType, extendedCloudCredential);
+            if (cpuCountForInstanceType.isPresent()) {
+                result = cpuCountForInstanceType.get() - BootstrapParams.WORKER_THREAD_REDUCTION_COUNT;
+            }
+        }
+        return Math.max(BootstrapParams.DEFAULT_WORKER_THREADS, result);
+    }
```
Comment (lacikaaa) [inline freeipa/src/main/java/com/sequenceiq/freeipa/service/BootstrapService.java:179]: this logic is mostly duplicated, can't we have the common part somewhere where it's shared? also I'm not sure this is really required for freeipa at all

File: orchestrator-salt/src/main/java/com/sequenceiq/cloudbreak/orchestrator/salt/SaltOrchestrator.java [hunk 2]
```diff
@@ -219,6 +229,18 @@ public void bootstrap(List<GatewayConfig> allGatewayConfigs, Set<Node> targets,
...
 
+    private void uploadSaltMasterConfig(SaltConnector sc, BootstrapParams params, Set<String> gatewayTargets, ExitCriteriaModel exitModel)
+            throws CloudbreakOrchestratorFailedException {
+        try {
+            String masterConfig = freeMarkerTemplateUtils.processTemplateIntoString(
+                    freemarkerConfiguration.getTemplate("templates/master.conf.ftl", StandardCharsets.UTF_8.name()),
+                    Map.of("workerThreads", params.getMasterWorkerThreads()));
+            uploadFileToTargets(sc, gatewayTargets, exitModel, "/etc/salt/master.d", "custom.conf", masterConfig.getBytes(StandardCharsets.UTF_8));
+        } catch (IOException | TemplateException e) {
+            throw new RuntimeException(e);
```
Comment (lacikaaa) [inline orchestrator-salt/src/main/java/com/sequenceiq/cloudbreak/orchestrator/salt/SaltOrchestrator.java:240]: please replace with some custom exception and also log it

**Review by lacikaaa — CHANGES_REQUESTED** [review]

**Review by lacikaaa — CHANGES_REQUESTED** [review]
please add a bit more detail to the commit message (not to the GH desc) like:
- when this change would take effect as it requires master restart
- what is the motivation and the expected result

File: core/src/main/java/com/sequenceiq/cloudbreak/core/bootstrap/service/ClusterBootstrapper.java [hunk 3]
```diff
@@ -502,4 +535,33 @@ private void validatePollingResultForCancellation(PollingResult pollingResult, S
...
+        if (stateComponent != null) {
+            String content = (String) stateComponent.getAttributes().getMap().getOrDefault(ComponentType.SALT_STATE.name(), "");
+            if (!content.isEmpty()) {
+                stateZip = Base64.decodeBase64(content);
+            }
+        }
+        return stateZip;
+    }
+
+    public ExtendedCloudCredential convertCredentialToExtendedCloudCredential(Credential credential, CloudPlatform cloudPlatform) {
```
[the remark this answers is not in the document]
Comment (foldik) [reply core/src/main/java/com/sequenceiq/cloudbreak/core/bootstrap/service/ClusterBootstrapper.java]: I deleted it. Thanks.

File: freeipa/src/main/java/com/sequenceiq/freeipa/service/BootstrapService.java (hunk 1 above)
Comment (foldik) [reply freeipa/src/main/java/com/sequenceiq/freeipa/service/BootstrapService.java]: Now it's only here.

**Review by lacikaaa — APPROVED** [review]

**Pull request merged**